src/system/user.cr

```diff
@@ -1,13 +1,24 @@
 require "c/pwd"
 require "c/sys/limits"
 
-struct User
+struct System::User
+
+  module Limits
```

Comment:
No need to introduce a `Limits` submodule. Put the constant into `System::User` directly. Same for Group and GID_MAX.